backend/server/auth_api.js

```diff
@@ -0,0 +1,113 @@
+const express = require('express');
+const authRouter = express.Router();
+const {pool} = require('./db');
+const passport = require("passport");
+const {passwordHash} = require('../hash');
+const {validateEmail} = require('../utils');
+
+// Registering a user
+authRouter.post('/register', async (req, res) => {
+    const {username, password, nickname} = req.body;
+
+    if (!validateEmail(username)) {
+        return res.status(400).json({msg: 'Email is not valid'});
+    };
+
+    if (password.length < 8) {
+        return res.status(400).json({msg: 'Password needs to be at least 8 characters'});
+    }
+
+    if (!nickname) {
+        return res.status(400).json({msg: 'Nickname must be specified'});
+    }
+
+    try {
+        const check = await pool.query('select * from users where username = $1', [username]);
+        if (check.rows.length > 0) {
+            return res.status(400).json({msg: 'Email already exist, choose a different email'});
+        }
+
+        const hashedPassword = await passwordHash(password, 10);
+        const timestamp = new Date(Date.now());
+        await pool.query('INSERT INTO users (username, nickname, password, created_at) VALUES ($1, $2, $3, $4) returning *', [username, nickname, hashedPassword, timestamp]);
+        return res.status(201).json({msg: 'Success creating user'});
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+// Login a user - local strategy
+authRouter.get("/login", (req, res) => {
+    res.status(401).json({msg: 'Authentication failed'});
+});
+
+
+// Get user profile page
+authRouter.get("/profile", async (req, res) => {
+    try {
+        const result = await pool.query('select * from users where id = $1', [req.user.id]);
```

Comment:
what if the user is not logged in? 
Please give an appropriate error message for that (or just move it to the `game_api.js` that has a middleware to verify that).